readinessProbe/readiness/server_test.go

```diff
@@ -0,0 +1,98 @@
+package readiness
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	v1 "github.com/openshift/ocs-osd-deployer/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("ManagedOCS Readiness Probe", func() {
+	// Define utility constants for object names and testing timeouts/durations and intervals.
+	const (
+		ManagedOCSName = "test-managedocs"
+		TestNamespace  = "default"
+
+		timeout  = time.Second * 10
+		duration = time.Second * 10
+		interval = time.Millisecond * 250
+	)
+
+	getResourceKey := func(obj runtime.Object) client.ObjectKey {
+		key, err := client.ObjectKeyFromObject(obj)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+		return key
+	}
+
+	When("managedocs reports its components as \"not ready\"", func() {
+		BeforeEach(func() {
+			ctx := context.Background()
+
+			managedOCS := &v1.ManagedOCS{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      ManagedOCSName,
+					Namespace: TestNamespace,
+				},
+			}
+
+			Expect(k8sClient.Create(ctx, managedOCS)).Should(Succeed())
+
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, getResourceKey(managedOCS), managedOCS)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
```

Comment:
This code is bassicly `waitForResource` from `managedocs_controller_test.go`.
As of the last comment, I suggest moving this shared functionality to  `test/util.go`